fsspec/tests/test_spec.py

```diff
@@ -124,6 +134,49 @@ def test_glob(test_path, expected):
         assert info == test_fs[name]
 
 
+@pytest.mark.parametrize(
+    ["test_paths", "expected"],
+    [
+        (
+            ("top_level/second_level", "top_level/sec*", "top_level/*"),
+            [
+                "top_level/second_level",
+                "top_level/second_level/date=2019-10-01",
+                "top_level/second_level/date=2019-10-01/a.parquet",
+                "top_level/second_level/date=2019-10-01/b.parquet",
+                "top_level/second_level/date=2019-10-02",
+                "top_level/second_level/date=2019-10-02/a.parquet",
+                "top_level/second_level/date=2019-10-04",
+                "top_level/second_level/date=2019-10-04/a.parquet",
+            ],
+        ),
+        (("misc/foo.txt", "misc/*.txt"), ["misc/foo.txt"]),
+        # Note: fails because 'glob_test' paths are missing (intentionally?) from ls,
+        #  remove comments and add id to fail it
+        # Todo: figure out what test_fs.ls("") ought to return
```

Comment:
I would think `["glob_test", "misc", "top_level"]`

We assume "" is not itself a directory, but it could be thought of as one.